core-bundle/src/Search/Backend/BackendSearch.php

```diff
@@ -143,8 +176,11 @@ private function convertSearchDocumentToProviderHit(array $document): Hit|null
         $document = Document::fromArray(json_decode($document['document'], true, 512, JSON_THROW_ON_ERROR));
         $hit = $fileProvider->convertDocumentToHit($document);
 
+        // The provider did not find any hit for it anymore so it must have been removed
+        // or expired. Remove from the index.
         if (!$hit) {
-            // TODO: delete this document from the index
+            $this->deleteDocuments([$hit->getDocument()]);
```

Comment:
This cannot work because `if (!$hit)`.

Reply:
Yes I saw that. Will write a unit test that covers this case.

Reply:
Fixed.